sources/packages/backend/apps/api/src/route-controllers/announcements/_tests_/e2e/announcement.institutions.controller.getAnnouncements.e2e-spec.ts

```diff
@@ -0,0 +1,127 @@
+import { addDays } from "@sims/utilities";
+import {
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getInstitutionToken,
+  InstitutionTokenTypes,
+} from "../../../../testHelpers";
+
+describe("AnnouncementInstitutionsController(e2e)-getAnnouncements", () => {
+  let app: INestApplication;
+  let announcementsRepo: Repository<Announcement>;
+  const endpoint = `/institutions/announcements?target=institution-dashboard`;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    announcementsRepo = dataSource.getRepository(Announcement);
+  });
+
+  it("Should return a current announcement.", async () => {
+    // Arrange
+    const now = new Date();
+    const before = addDays(-10, now);
+    const future = addDays(30, now);
+
+    const announcement = new Announcement();
+    announcement.message = "test announcement";
+    announcement.messageTitle = "test title";
+    announcement.target = ["student-dashboard", "institution-dashboard"];
+    announcement.startDate = before;
+    announcement.endDate = future;
+    await announcementsRepo.save(announcement);
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+
+    // Act/Assert
+    const response = await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK);
+
+    expect(
+      response.body.some(
+        (announcement) =>
+          announcement.message === "test announcement" &&
+          announcement.messageTitle === "test title" &&
+          announcement.startDate === before.toISOString() &&
+          announcement.endDate === future.toISOString(),
+      ),
+    ).toBe(true);
+  });
+
+  it("Should not return an outdated announcement.", async () => {
+    // Arrange
+    const now = new Date();
+    const before = addDays(-20, now);
+
+    const announcement = new Announcement();
+    announcement.message = "already done";
+    announcement.messageTitle = "already done";
+    announcement.target = ["student-dashboard", "institution-dashboard"];
+    announcement.startDate = before;
+    announcement.endDate = before;
+    await announcementsRepo.save(announcement);
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+
+    // Act/Assert
+    const response = await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK);
+
+    expect(
+      response.body.some(
+        (announcement) =>
+          announcement.message === "already done" &&
+          announcement.messageTitle === "already done" &&
+          announcement.startDate === before.toISOString() &&
+          announcement.endDate === before.toISOString(),
+      ),
+    ).toBe(false);
+  });
+
+  it("Should not return a future announcement.", async () => {
```

Comment:
For this test, we can either clear the data for "student-dashboard" before or use a fake target to test the outdated message. 

```ts
  // Arrange
    const now = new Date();
    const before = addDays(-20, now);

    const announcement = new Announcement();
    announcement.message = "already done";
    announcement.messageTitle = "already done";
    announcement.target = ["outdated-dashboard"];
    announcement.startDate = before;
    announcement.endDate = before;
    await announcementsRepo.save(announcement);
    const institutionUserToken = await getInstitutionToken(
      InstitutionTokenTypes.CollegeFUser,
    );

    const endpointWithOutdatedDashboard = `/institutions/announcements?target=institution-dashboard`;

    // Act/Assert
    await request(app.getHttpServer())
      .get(endpointWithOutdatedDashboard)
      .auth(institutionUserToken, BEARER_AUTH_TYPE)
      .expect(HttpStatus.OK)
      .expect([]);
  });
```